ruby/ext/google/protobuf_c/repeated_field.c

```diff
@@ -47,6 +47,15 @@ RepeatedField* ruby_to_RepeatedField(VALUE _self) {
   return self;
 }
 
+int index_position(VALUE _index, RepeatedField* repeated_field){
+  int index = NUM2INT(_index);
+  if (index < 0 && repeated_field->size > 0) {
+    index = repeated_field->size + index;
+  }
+  return index;
```

Comment:
I did not add this to the header file; it has been awhile but if I remember correctly header files are really for methods you want exposed vs internal-use-only methods.

anyway, let me know if there are any issues here, including styling and naming conventions